### PR TITLE
Feat task12 collectionandgenerics

### PR DESCRIPTION
**Pull request by AncyJenisha**
Different types of collections and operations on them with generics.

### CODE REVIEW

File: src/Assignment12/CollectionAndGenerics/GenericDictionary.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    /// <typeparam name="TKey">Generic type for the Student name</typeparam>
+    /// <typeparam name="TValue">Generic type for student grade.</typeparam>
+    public class GenericDictionary<TKey, TValue>
+         where TKey : notnull
+    {
+        /// <summary>
+        /// Gets or sets the student details to the generic dictionary.
+        /// </summary>
+        /// <value>Generic type of student name and grade.</value>
+        public static Dictionary<TKey, TValue> GenericDictionaryOfStudentDetails { get; set; } = new ();
```
Comment (krrish-soliton) [inline src/Assignment12/CollectionAndGenerics/GenericDictionary.cs]: Can you name it StudentDetails?

File: src/Assignment12/CollectionAndGenerics/GenericDictionary.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        /// <summary>
+        /// Gets the count of Generic Dictionary.
+        /// </summary>
+        /// <value>Key as string and value as float</value>
+        public int GenericDictionaryCount => GenericDictionaryOfStudentDetails.Count;
```
Comment (krrish-soliton) [inline src/Assignment12/CollectionAndGenerics/GenericDictionary.cs]: This should be Count

File: src/Assignment12/CollectionAndGenerics/GenericList.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    /// Class with methods for generic list.
+    /// </summary>
+    /// <typeparam name="T">Generic Type parameter.</typeparam>
+    public class GenericList<T>
+    {
+        /// <summary>
+        /// Gets or sets the books list in generic type.
+        /// </summary>
+        /// <value>Book name as String</value>
+        public static List<T> GenericListOfBooks { get; set; } = new ();
```
Comment (krrish-soliton) [inline src/Assignment12/CollectionAndGenerics/GenericList.cs]: Name it as Books.

File: src/Assignment12/CollectionAndGenerics/GenericQueue.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    /// Has methods for operations on GenericQueue
+    /// </summary>
+    /// <typeparam name="T">Generic Type parameter.</typeparam>
+    public class GenericQueue<T>
+    {
+        /// <summary>
+        /// Gets or Sets the Names as string to the generic list.
+        /// </summary>
+        /// <value>Names as String.</value>
+        public static Queue<T> GenericQueueOfNames { get; set; } = new Queue<T>();
```
Comment (krrish-soliton) [inline src/Assignment12/CollectionAndGenerics/GenericQueue.cs]: Make it as Names.

**Review by krrish-soliton — APPROVED** [review]

File: src/Assignment12/CollectionAndGenerics/GenericDictionary.cs (hunk 1 above)
Comment (AncyJenisha) [reply src/Assignment12/CollectionAndGenerics/GenericDictionary.cs]: Variable name changed to StudentDetails.

File: src/Assignment12/CollectionAndGenerics/GenericDictionary.cs (hunk 2 above)
Comment (AncyJenisha) [reply src/Assignment12/CollectionAndGenerics/GenericDictionary.cs]: Variable name changed to count.

File: src/Assignment12/CollectionAndGenerics/GenericQueue.cs (hunk 4 above)
Comment (AncyJenisha) [reply src/Assignment12/CollectionAndGenerics/GenericQueue.cs]: Variable name changed to Names.

File: src/Assignment12/CollectionAndGenerics/GenericList.cs (hunk 3 above)
Comment (AncyJenisha) [reply src/Assignment12/CollectionAndGenerics/GenericList.cs]: variable name changed to books

**Pull request merged**